### DISCUSSION TITLE
Circuit Labels / Container Names / Hover Descriptions

### DISCUSSION THREAD

**Discussion by Saevon** [FakeFishGames/Barotrauma#4714] [Ideas and suggestions]
# Problem

Currently rewiring things is pretty confusing. Especially on a ship you've never been on. If most of the wires had better naming (automatic or otherwise) it would be easier to rewire in the middle of a game (or even an emergency).

Item storage can have a similar problem.

# Solution

## Container Labels

Being able to label storage makes sorting easier.  Both in the sub editor, and renaming in-game (for the crew to customize storage)

This would show similar to "Supplies Cabinet" when you open up the storage. But it would also show on hover, replacing the current label `Supplies Cabinet` with `Medkit` or `EVA Gear`, etc.

<details>
  <summary><strong>Examples</strong></summary>

  You would change the text when the container is open
<img width="326" alt="Screen Shot 2021-01-17 at 1 30 50 AM" src="https://user-images.githubusercontent.com/570036/104836591-c1b80000-5863-11eb-8fac-fc2a201a112b.png">

But also the text when hovering
<img width="248" alt="Screen Shot 2021-01-17 at 1 30 38 AM" src="https://user-images.githubusercontent.com/570036/104836590-c086d300-5863-11eb-8b42-39aef7fb57ec.png">

</details>

## Wiring Component Labels

It would be much easier for an engineer (or other jobs) to mess with electronics if they can label machinery/items.

1. If you have junction boxes, then it's hard to know what to prioritize repairing.  If the junction was called `Pump Junction Box` and this was visible: on hover, in the power-status (on click), in the rewiring page, etc, this would be much easier. You can also pull power for individual things easier (temporarily)
2. If you have logic, then a junction usually leads to `relay (power_in)` rather then `Auto Oxygen relay (power_in)` which would be much easier to re-route in game (in the editor as well).   So you would "rename" the relay to be called `Auto Oxygen relay`
3. Wifi Components make this harder as well. Being able to "rename" `Wifi Component` to `Water Data [Wifi]` would make this much easier.

P.S. Junction Boxes could be super helpful if they're actually "Circuit Breakers", with a switch on the "click" interface disabling all power / signal transmission.  (You can use a relay + switch as well, or just unwire the source cable. So this isn't vital)

**Comment by mLuby** [comment]
This would be super helpful. But we have to remember that this game has an international audience, so translations are important. Any idea how this could handle that?

**Reply by Saevon** [reply]
Great Point!

Currently the way it works for hulls, is there are a bunch of common names that can be generated. Based on the position in the ship.

I would expect a similar common list, perhaps we as the community could come up with one?
 * Containers should be easiest, there's a pretty finite "common" list
 * Circuits are specifically made for creativity, so I'm not sure it doable…
 * Buttons & Switches might have common uses?  I wonder if we do a submarine survey we could see what "weird" button uses are actually common and might be added to a translation list.

**Reply by JamesTCrenshaw** [reply]
What might be better is to use pictograms. they wouldn't need to be translated, and using simple images like lightbulbs and a ships' wheel would be able to communicate locations or parts of the ship.

**Reply by Saevon** [reply]
Parts of the ship in the labels could be good for Junction Boxes.

**Reply by mLuby** [reply]
So kind of like these? (Sorry for low resolution.)
![tmp](https://user-images.githubusercontent.com/2483420/104965710-4ebaa080-59ad-11eb-8f78-a755ce3cfe43.png)

**Reply by Saevon** [reply]
Sorta, I assumed this would be for the circuits labels part (aka the wires heading into the junction box would be labelled. So I was imagining something like this:

<img width="920" alt="Screen Shot 2021-01-18 at 2 02 51 PM" src="https://user-images.githubusercontent.com/570036/104966712-77836b80-5996-11eb-92c6-8fd52d447123.png">

Which would help a lot if that went through a relay first, (The relay could also be labelled). Or if it was an input controlling a relay.
